cmake/HPX_AddPkgConfigExternalDepsTarget.cmake

```diff
@@ -0,0 +1,13 @@
+# Copyright (c) 2019 The STE||AR-Group
+#
+# Distributed under the Boost Software License, Version 1.0. (See accompanying
+# file LICENSE_1_0.txt or copy at http://www.boost.org/LICENSE_1_0.txt)
+
+# This files is including the external targets which are imported by the modules
+# as they are not available in the main CMakeLists.txt scope, those targets
+# exist in the module scope.
+# We decided to leave it that way and not make the
+# include(HPX_BoostProgramOptions) global to the whole HPX project because it
+# seems more clean
+include(HPX_SetupBoostProgramOptions)
+include(HPX_SetupBoostFilesystem)
```

Comment:
Shouldn't this be module-specific (and `PUBLIC`)?

Reply:
This is special handling for pkgconfig (which is why it comes last in the main `CMakeLists.txt` so that nothing else sees it). Targets set up in the modules are not visible at the top level where they're needed for the pkgconfig file generation.